src/_pytest/raises_group.py

```diff
@@ -0,0 +1,1211 @@
+            # TODO: it instructs to use `-v` to print leading text, but that doesn't work
+            # I also don't know if this is the proper entry point, or tool to use at all
+            from _pytest.assertion.util import _diff_text
+            from _pytest.assertion.util import dummy_highlighter
+
+            diff = _diff_text(self.rawmatch, stringified_exception, dummy_highlighter)
+            self._fail_reason = ("\n" if diff[0][0] == "-" else "") + "\n".join(diff)
+            return False
+
+        # I don't love "Regex"+"Input" vs something like "expected regex"+"exception message"
+        # when they're similar it's not always obvious which is which
+        self._fail_reason = (
+            f"Regex pattern did not match{maybe_specify_type}.\n"
+            f" Regex: {_match_pattern(self.match)!r}\n"
+            f" Input: {stringified_exception!r}"
+        )
+        if _match_pattern(self.match) == stringified_exception:
+            self._fail_reason += "\n Did you mean to `re.escape()` the regex?"
+        return False
+
+    @abstractmethod
+    def matches(
+        self: AbstractRaises[BaseExcT_1], exception: BaseException
+    ) -> TypeGuard[BaseExcT_1]:
+        """Check if an exception matches the requirements of this AbstractRaises.
+        If it fails, :meth:`AbstractRaises.fail_reason` should be set.
+        """
+
+
+@final
+class RaisesExc(AbstractRaises[BaseExcT_co_default]):
+    """
+    .. versionadded:: 8.4
+
+    Helper class to be used together with RaisesGroup when you want to specify requirements on sub-exceptions.
+
+    You don't need this if you only want to specify the type, since :class:`RaisesGroup`
+    accepts ``type[BaseException]``.
+
+    The type is checked with :func:`isinstance`, and does not need to be an exact match.
+    If that is wanted you can use the ``check`` parameter.
+
+    :meth:`RaisesExc.matches` can also be used standalone to check individual exceptions.
+
+    Examples::
+
+        with RaisesGroup(RaisesExc(ValueError, match="string"))
+            ...
+        with RaisesGroup(RaisesExc(check=lambda x: x.args == (3, "hello"))):
+            ...
+        with RaisesGroup(RaisesExc(check=lambda x: type(x) is ValueError)):
+            ...
+    """
+
+    # Trio bundled hypothesis monkeypatching, we will probably instead assume that
+    # hypothesis will handle that in their pytest plugin by the time this is released.
+    # Alternatively we could add a version of get_pretty_function_description ourselves
+    # https://github.com/HypothesisWorks/hypothesis/blob/8ced2f59f5c7bea3344e35d2d53e1f8f8eb9fcd8/hypothesis-python/src/hypothesis/internal/reflection.py#L439
+
+    # At least one of the three parameters must be passed.
+    @overload
+    def __init__(
+        self: RaisesExc[BaseExcT_co_default],
+        expected_exception: (
+            type[BaseExcT_co_default] | tuple[type[BaseExcT_co_default], ...]
+        ),
+        match: str | Pattern[str] | None = ...,
```

Comment:
Let's make `match` and `check` kw-only arguments (also in the real `__init__` below)

Reply:
Fixed in https://github.com/pytest-dev/pytest/pull/13279/commits/af1c472aaf6080272dce6e3b9826e61d954df5cf in #13279